sdk/python/kfp/compiler/pipeline_spec_builder.py

```diff
@@ -691,11 +702,20 @@ def build_component_spec_for_group(
             component_spec.input_definitions.artifacts[
                 input_name].is_artifact_list = channel.is_artifact_list
         else:
-            # channel is one of PipelineParameterChannel, LoopArgument, or
-            # LoopArgumentVariable.
-            component_spec.input_definitions.parameters[
-                input_name].parameter_type = type_utils.get_parameter_type(
-                    channel.channel_type)
+            # channel is one of PipelineParameterChannel, LoopParameterArgument,
+            # LoopArtifactArgument, or LoopArgumentVariable.
+            if isinstance(channel, for_loop.LoopArtifactArgument):
+                component_spec.input_definitions.artifacts[
+                    input_name].artifact_type.CopyFrom(
+                        type_utils.bundled_artifact_to_artifact_proto(
+                            channel.channel_type))
+
+                component_spec.input_definitions.artifacts[
+                    input_name].is_artifact_list = channel.is_artifact_list
+            else:
+                component_spec.input_definitions.parameters[
+                    input_name].parameter_type = type_utils.get_parameter_type(
+                        channel.channel_type)
```

Comment:
See comment above about flattening nested conditions. Is that possible here?

Reply:
Done.